src/drivers/cmakeDriver.ts

```diff
@@ -1696,6 +1714,34 @@ export abstract class CMakeDriver implements vscode.Disposable {
         }
     }
 
+    private async _findBuildTask(): Promise<vscode.Task | undefined> {
```

Comment:
It looks like that `_findBuildTask` is an implementation of `vscode.commands.executeCommand("workbench.action.tasks.build");` , it fetches all of the tasks, choose the default is existed, otherwise shows a quickpick to choose from. 
The only difference is that `vscode.commands.executeCommand("workbench.action.tasks.build")` will show the template tasks besides the tasks defined in `tasks.json`.

Reply:
That's right. However, it is still not delivering the results we want. Please read the later comments.